### PR TITLE
Add unit tests for censors, housekeeping

### PR DESCRIPTION
**Pull request by nwithan8**
- Improvements to censors testing (this got lost in work a few months ago)
- Ignore cassettes in git (they all get deleted during unit tests, so no need for them to be in this repo and kept updated)
- Bug fix for base URL and full URL matching

### CODE REVIEW

**Comment by Justintime50** [comment]
Looks like we need to fix these tests

**Comment by nwithan8** [comment]
> Looks like we need to fix these tests

Works on my machine, even with a clean install...

File: EasyVCR.Tests/CensorsTest.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+{
+    [TestClass]
+    public class CensorsTest
+    {
+        [TestMethod]
+        public void TestDefaultCensors()
+        {
+            var censors = Censors.Default;
+
+            Assert.IsTrue(true);
```
Comment (Justintime50) [inline EasyVCR.Tests/CensorsTest.cs:14]: question: what does this test actually do? We aren't making any valid assertions here

File: EasyVCR.Tests/CensorsTest.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+        [TestMethod]
+        public void TestApplyBodyParametersCensorsNoContentType()
+        {
+            const string body = "{\"foo\":\"bar\"}";
+            InternalUtilities.ContentType? contentType = null;
+
+            var censors = new Censors();
+
+            Assert.ThrowsException<VCRException>(() => censors.ApplyBodyParametersCensors(body, contentType));
```
Comment (Justintime50) [inline EasyVCR.Tests/CensorsTest.cs:25]: question: why do we expect an error here?

hint: docstring could easily fix this question

File: EasyVCR.Tests/CensorsTest.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+            var censors = new Censors();
+
+            var result = censors.ApplyPathElementsCensors(url);
+
+            Assert.AreEqual(url, result);
+        }
+
+        [TestMethod]
+        public void TestApplyPathElementsCensorsNoCensorsReturnsOriginalUrl()
```
Comment (Justintime50) [inline EasyVCR.Tests/CensorsTest.cs:154]: question: we appear to be testing a bunch of the "empty" to "original" flows, are we already testing the "filled" flows elsewhere?

Comment (nwithan8) [reply EasyVCR.Tests/CensorsTest.cs:194]: Yes, we use censors in the unit tests for other functionality, which doesn't seem to interfere with their operation.

This is a bunch of happy and sad path testing for censors specifically.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## Next Release
+
+- Fix a bug where URLs were not be extracted correctly, potentially causing false matches when matching by URL
```
Comment (Justintime50) [inline CHANGELOG.md:5]:
```suggestion
- Fix a bug where URLs were not being extracted correctly, potentially causing false matches when matching by URL
```

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**